server/task/src/main/scala/org/apache/james/task/eventsourcing/EventSourcingTaskManager.scala

```diff
@@ -21,19 +21,19 @@ package org.apache.james.task.eventsourcing
 import java.io.Closeable
 import java.util
 
-import com.google.common.annotations.VisibleForTesting
 import javax.inject.Inject
+
 import org.apache.james.eventsourcing.eventstore.{EventStore, History}
-import org.apache.james.eventsourcing.{AggregateId, Subscriber}
+import org.apache.james.eventsourcing.{AggregateId, EventSourcingSystem, Subscriber}
 import org.apache.james.task._
 import org.apache.james.task.eventsourcing.TaskCommand._
 
+import com.google.common.annotations.VisibleForTesting
+
 import scala.annotation.tailrec
 
 class EventSourcingTaskManager @Inject @VisibleForTesting private[eventsourcing](
-                                                                                  val listener: WorkerStatusListener,
-                                                                                  val worker: TaskManagerWorker,
-                                                                                  val workQueue: WorkQueue,
+                                                                                  workQueueSupplier: EventSourcingSystem => WorkQueue,
```

Comment:
what is the point of passing a supplier to only call it once?

Reply:
dependency problem